src/selectors.ts

```diff
@@ -184,7 +185,7 @@ export class Selectors {
       if (eqIndex !== -1 && part.substring(0, eqIndex).trim().match(/^[a-zA-Z_0-9-+:]+$/)) {
         name = part.substring(0, eqIndex).trim();
         body = part.substring(eqIndex + 1);
-      } else if (part.startsWith('"')) {
+      } else if (hasTextSelectorSurroundings(part)) {
         name = 'text';
         body = part;
```

Comment:
Now that you brought unpleasant JSON.parse error, perhaps we should try to JSON.parse right here in node to get a nice stack trace and/or error message? I'd also add a test for that.